clang/arm: Do not use MUST_TAIL_CALL

This causes clang-17 to crash see [1]
this code is new in webkit 2.42[2] thats why we do not see the crash in older webkit

[1] https://github.com/llvm/llvm-project/issues/67767
[2] https://github.com/WebKit/WebKit/commit/4d816460b765acd8aef90ab474615850b91ecc35

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
--- a/Source/WTF/wtf/Compiler.h
+++ b/Source/WTF/wtf/Compiler.h
@@ -284,7 +284,7 @@
 /* MUST_TAIL_CALL */
 
 #if !defined(MUST_TAIL_CALL) && defined(__cplusplus) && defined(__has_cpp_attribute)
-#if __has_cpp_attribute(clang::musttail)
+#if __has_cpp_attribute(clang::musttail) && !defined(__arm__)
 #define MUST_TAIL_CALL [[clang::musttail]]
 #endif
 #endif
